lib/IdPDisco.php

```diff
@@ -83,11 +83,19 @@ public function handleRequest()
 
         // Get the SP's name
         $spEntries = Metadata::getSpMetadataEntries($metadataPath);
-        $spName = $spEntries[$spEntityId][self::$spNameMdKey] ?? null;
 
         $templateFileName = 'selectidp-' . $this->config->getString('idpdisco.layout', 'links') . '.php';
 
         $t = new SimpleSAML_XHTML_Template($this->config, $templateFileName, 'disco');
+
+        $spName = null;
+
+        $rawSPName = $spEntries[$spEntityId][self::$spNameMdKey] ?? null;
+        if ($rawSPName !== null) {
+            $spName = htmlspecialchars($t->getTranslator()->getPreferredTranslation(
+                SimpleSAML\Utils\Arrays::arrayize($rawSPName, 'en')))   ;
```

Comment:
For PSR-2 formatting, I would recommend changing...
```php
            $spName = htmlspecialchars($t->getTranslator()->getPreferredTranslation(
                SimpleSAML\Utils\Arrays::arrayize($rawSPName, 'en')))   ;
```
... to...
```php
            $spName = htmlspecialchars($t->getTranslator()->getPreferredTranslation(
                SimpleSAML\Utils\Arrays::arrayize($rawSPName, 'en')
            ));
```
... but that's just for readability, not to fix a bug.

Reply:
Fixed.